python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1595,43 +1595,35 @@ def _get_failed_asset_partitions(
     result: List[AssetKeyPartitionKey] = []
 
     for run in runs:
+        planned_asset_keys = instance_queryer.get_planned_materializations_for_run(
+            run_id=run.run_id
+        )
+        completed_asset_keys = instance_queryer.get_current_materializations_for_run(
+            run_id=run.run_id
+        )
```

Comment:
i am a little confused what happens here in the case where a chunked backfill run materializes some partitions, but then fails before it materializes all of them. the asset key that failed will be included in completed_asset_keys still, right? Do we correctly account for the missing partitions in that case?

Reply:
Hmm I would not be surprised if we don't correctly account for it.

FWIW I'd expect it to be a rare case, because user code can't yield partition materializations individually. So it would only happen if there's a failure in the middle of the framework loop that stores the partition materializations.

Reply:
In theory single-run backfill does not have the capacity to fail on a subset of partitions for a given key. In general, the operation that it's modeling is transactional, the common one being "update this range of dates in a database table using a sql query". Under the hood, that operation is mapped to a single `Output` event, which is then in turn mapped to a `AssetMaterialization` event for each partition (so the Output is a single event which is directly associated with N partitions).

That being said, in theory it's possible for the act of emitting those AssetMaterialization events to fail, and I'm not sure what weirdness that would cause in practice if that happened (which I would imagine would be a pretty rare scenario).

But long story short I think the generalization of "if an asset has been materialized at all for this step, then all of its partitions have been materialized" is mostly true.